Polyhedron/demo/Polyhedron/Plugins/Surface_mesh_deformation/Scene_edit_polyhedron_item.cpp

```diff
@@ -1041,7 +1049,10 @@ void Scene_edit_polyhedron_item_priv::remesh(Mesh* mesh)
   unsigned int nb_iter = ui_widget->remeshing_iterations_spinbox->value();
 
   std::cout << "Remeshing (target edge length = " << target_length <<")...";
-  Is_constrained_map<Mesh> icm(&constrained_vec, mesh);
+  SMesh::Property_map<sm_vertex_descriptor,int> smap;
+  std::map<poly_vertex_descriptor,int> polymap;
+
+  Is_constrained_map<Mesh> icm(&constrained_vec,&polymap, &smap,  mesh);
```

Comment:
That API is really strange. `smap` and `polymap` should remain internals of the property map. For the internals
 - of `Is_constrained_map<Poiyhedron>`, you can use `std::shared_ptr<std::map<poly_vertex_descriptor,int>>` and the corresponding `std::make_shared`, to let C++ deal with the handling of memory,
 - and for `Is_constrained_map<SMesh>`, you could also use `std::shared_ptr<SMesh::Property_map<sm_vertex_descriptor,int> >` and create the shared pointer with a lambda-function as the `Deleter`, that would execute `sm_item->polyhedron()->remove_property_map(smap)` once that resource is not longer used.